pandas/core/generic.py

```diff
@@ -128,15 +128,17 @@
 # goal is to be able to define the docs close to function, while still being
 # able to share
 _shared_docs = {**_shared_docs}
-_shared_doc_kwargs = {
-    "axes": "keywords for axes",
-    "klass": "Series/DataFrame",
-    "axes_single_arg": "int or labels for object",
-    "args_transpose": "axes to permute (int or label for object)",
-    "optional_by": """
-        by : str or list of str
-            Name or list of names to sort by""",
-}
+_shared_doc_kwargs.update(
```

Comment:
don't use update, instead copy the dict & then update (otherwise you are mutating global state)